examples/vision/metric_learning_tf_similarity.py

```diff
@@ -0,0 +1,432 @@
+other and "dissimilar" inputs are pushed farther apart. Once trained, these
+models can produce embeddings for downstream systems where such similarity is
+useful, e.g., as a ranking signal for search or as a form of pretrained
+embedding model for another supervised problem.
+
+For a more detailed overview of metric learning see:
+
+* [What is metric learning?](http://contrib.scikit-learn.org/metric-learn/introduction.html)
+* ["Using crossentropy for metric learning" tutorial](https://www.youtube.com/watch?v=Jb4Ewl5RzkI)
+"""
+
+"""
+## Setup
+
+This tutorial will use [Tensorflow Similarity](https://github.com/tensorflow/similarity)
+to learn and evaluate the similarity embedding. The tensorflow similarity
+package provides a number of components that:
+
+* Make training contrastive models simple and fast.
+* Make it easier to ensure that batches contain pairs of examples.
+* Enable the evaluation of the quality of the embedding.
+"""
+
+import random
+
+from matplotlib import pyplot as plt
+from mpl_toolkits.axes_grid1 import ImageGrid
+import numpy as np
+
+import tensorflow as tf
+from tensorflow.keras import layers
+from tensorflow.keras.optimizers import Adam
+
+import tensorflow_similarity as tfsim
+from tensorflow_similarity.layers import MetricEmbedding  # row wise L2 norm
+from tensorflow_similarity.losses import (
+    MultiSimilarityLoss,
+)  # specialized similarity loss
+from tensorflow_similarity.models import (
+    SimilarityModel,
+)  # TF model with additional features
+from tensorflow_similarity.samplers import (
+    TFDatasetMultiShotMemorySampler,
+)  # Get and samples TF dataset catalog
+from tensorflow_similarity.visualization import confusion_matrix  # matching performance
+from tensorflow_similarity.visualization import (
+    projector,
+)  # allows to interactively explore the samples in 2D space
+from tensorflow_similarity.visualization import (
+    viz_neigbors_imgs,
+)  # neighbors visualization
+
+
+tfsim.utils.tf_cap_memory()
+
+print("TensorFlow:", tf.__version__)
+print("TensorFlow Similarity", tfsim.__version__)
+
+"""
+## Dataset Samplers
+
+We will be using the
+[CIFAR-10](https://www.tensorflow.org/datasets/catalog/cifar10) dataset for this
+tutorial.
+
+For a similarity model to learn efficiently, each batch must contains at least 2
+examples of each class.
+
+To make this easy, tf_similarity offers `Samplers()` that enable you to set both
+the number of classes and the minimum number of examples of each class per
+batch.
+
+The train and test datasets will be created using the
+`TFDatasetMultiShotMemorySampler()`. This creates a sampler that loads datasets
+from [tensorflow datasets](https://www.tensorflow.org/datasets) and yields
+batches containing a target number of classes and a target number of examples
+per class. Additionally, we can restrict the sampler to only yield the subset of
+classes defined in class_list, enabling us to train on a subset of the classes
```

Comment:
Use backticks around code keywords (we'll do a separate round of copyediting at the end)

Reply:
Done. Went through the rest of the document as well to try catch any other code keywords.